*: improve cluster conditions

hongchaodeng opened the pull request:
ref: https://github.com/coreos/etcd-operator/issues/1388

hasbro17 commented on pkg/apis/etcd/v1beta2/status.go:
```diff
@@ -0,0 +1,180 @@
+// If a cluster is upgrading and encountered a problem that prevents the upgrade,
+// the Upgrading condition's status will would be False and communicate the problem back.
+type ClusterCondition struct {
+	// Type of deployment condition.
```
Type of **cluster** condition

hongchaodeng commented:
fixed

hasbro17 commented:
Currently we're not removing any of the temporary conditions(like `Scaling`) after they finish successfully right?

For e.g if the operator finishes scaling the cluster from 3 to 5, then the `Scaling` condition will still be visible as set to True with the message `Scaling up: Current cluster size: 3, desired cluster size: 5`.

hasbro17 commented on pkg/apis/etcd/v1beta2/status.go:
```diff
@@ -0,0 +1,192 @@
+	// ControlPuased indicates the operator pauses the control of the cluster.
+	ControlPaused bool `json:"controlPaused"`
+
+	// Condition keeps ten most recent cluster conditions
```
Condition keeps track of all cluster conditions, if they exist.

hongchaodeng replied:
fixed

hongchaodeng commented:
@hasbro17 
The primary purpose of this PR is to add the Conditions according to the design doc. On changing that, there are some places to be fixed.

Nonetheless, some of them are pretty aligned with our expectation of Conditions, so I changed them; but some wasn't, so I removed them:

- Scaling condition was retained as True status
- The RemoveDeadMemberCondition was removed because it would come into event category.
- Upgrading condition was retained as True status
- Ready is changed to Available Condition as True status.

There are more work to be done. This would only keep them as it and try to layout the new Condition definition only.

hasbro17 commented:
Sounds good. LGTM.
